modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2508,10 +2509,13 @@ def from_arrow(cls, at, index_cols=None, index=None, columns=None):
             new_columns = pd.Index(data=at.column_names, dtype="O")
             new_index = pd.RangeIndex(at.num_rows)
 
-        new_dtypes = pd.Series(
-            [cls._arrow_type_to_dtype(col.type) for col in at.columns],
-            index=at.column_names,
-        )
+        new_dtypes = []
+
+        for idx, col in enumerate(at.columns):
+            if pyarrow.types.is_dictionary(col.type):
+                new_dtypes.append(LazyProxyCategoricalDtype(at, idx))
```

Comment:
Why the table will be outdated? A new frame could be created with a new table, but the old one will still be valid. The only issue here is that we keep reference to the old table, that is not used any more and could be garbage collected.

Updatable reference to frame does not solve it. For example, a new, temporary frame created and abandoned. We have updated the reference and now the temporary frame could not be garbage collected. Besides, I'm not sure if the arrow table could be missing in case of fall back to pandas, when a pandas frame is stored in partitions.

I think, a better solution is to clone LazyProxyCategoricalDtype when a new frame is created with a different table.